File: week-02/day-1/src/Anagram.java
import java.util.Arrays;

public class Anagram {
    public static void main(String[] args) {
//        Create a function named anagram following your current language's style guide.
//        It should take two strings and return a boolean value depending on whether its
//        an anagram or not.
//
//                  Examples
//        input 1	input 2	output
//        "dog" 	"god"	 true
//        "green"   "fox"	 false

        String sample1 = "dog";
        String sample2 = "god";
        boolean isAnagram = anagram(sample1, sample2);

        System.out.println(isAnagram);
    }

    private static boolean anagram(String sample1, String sample2) {
        boolean result = false;

        // split word by character
        String[] firstWord = sample1.split("");
        String[] secondWord = sample2.split("");

        // sort the arrays
        Arrays.sort(firstWord);
        Arrays.sort(secondWord);

        if (Arrays.equals(firstWord, secondWord)) {
            return true;
        }
        return result;
    }
}

File: python_solutions/week-02/day-1/functions/anagram.py
# Exercise
# Create a function named is_anagram following your current language's style
# guide. It should take two strings and return a boolean value depending on
# whether its an anagram or not.

# Examples
# | input 1 | input 2 | output |
# | :-----: | :-----: | :----: |
# |  "dog"  |  "god"  |  true  |
# | "green" |  "fox"  |  false |


def are_anagrams(string_1: str, string_2: str):
    intersection = set(list(string_1)).intersection(set(list(string_2)))
    return True if len(intersection) is len(string_1) and \
                   len(intersection) is len(string_2) else False


if __name__ == '__main__':
    print(are_anagrams("dog", "god"))
    print(are_anagrams("green", "fox"))
    print(are_anagrams("asd", "dfg"))
